nikola/plugins/task/page_index.py

```diff
@@ -64,6 +64,8 @@ def is_enabled(self, lang=None):
     def classify(self, post, lang):
         """Classify the given post for the given language."""
         destpath = post.destination_path(lang, sep='/')
+        if post._has_pretty_url(lang):
+            destpath = destpath.rstrip('/index.html')
```

Comment:
If, for whatever obscure reason, your path happens to be `https://example.com/something/index.html/index.html/index.html`, this will not result in what you want ;-)